### ISSUE TITLE
Implement VisualStudio 2017 install

### ISSUE THREAD

**Issue by avishnyakov** [SubPointSolutions/metabox#13]
As a metabox user, I want to be able to use VisualStudio 2017 out of the box. 

Currently, metabox installs only VS2013 and VS2015. Both installs are similar, made to support SharePoint related development and converted with Pester testing.

VisualStudio 2017 has got a different installation flow. From what we can see, a local cache needs to be build before installation. With metabox, a file resource should be implemented to get VS2017 cache build and packed into ZIP files for the further transfer into Vagrant VMs. Installation script and Ruby handler needs to be implemented as well.

The aim is to provide a smooth, CI/CD-ed experience for VS2017 installation.

**Comment by avishnyakov** [comment]
Does anyone want to take this?

**Comment by shurick81** [comment]
It seems that there is no single file installer publicly available for VS 17.

They call it "Install Visual Studio 2017 on low bandwidth or unreliable network environments":
https://docs.microsoft.com/en-us/visualstudio/install/install-vs-inconsistent-quality-network

1. Download the Visual Studio bootstrapper
2. run bootstrapper file for creating "local layout": `vs_community.exe --layout c:\vs2017layout`

Seems like in general it wouldn't hear to have a windows VM that can run some MB system jobs:
1. Prepare SP slipstreamed media with AutoSPSourceBuilder.
2. Prepare VS 2017 installer.
3. Preparing some certificates.

Then these files can be stored on the virtual drive and can be connected to one VM or another.

**Comment by avishnyakov** [comment]
> Seems like in general it wouldn't hear to have a windows VM that can run some MB system jobs:
Yep, this prevents file downloads on mac/linux, on CI server. One should be either on win, or have a pre-downloaded layout. That is also why we automated VS2013/2015 and chose 2015 version as a default one.

**Comment by shurick81** [comment]
One more option to go is to download this resource from the Internet right before installing. It's less than 2GB package. This also connected to the issue of having a VS image. Currently, we use just Windows image and install VS on top of it during stack provisioning phase.

**Comment by SubPointSupport** [comment]
Updated link to new VS installed seems to be this one:
> https://download.visualstudio.microsoft.com/download/pr/12221250/52257ee3e96d6e07313e41ad155b155a/vs_Enterprise.exe

Adding into metabox file resource with further VS download as per layout file.

**Comment by shurick81** [comment]
Encountered an issue with downloading. Steps to reproduce:
1. delete `metabox_downloads\vs2017.vs_ent` directory
2. run `cls && "config/.metabox.downloads.bat" && rake resource:generate && rake resource:list && rake fileset:download[visualstudio::vs2017.vs_enterprise.exe]` from `integration` branch.

Expected: downloaded VS media to `vs2017.vs_ent` directory.
Actually: only few files are saved in `vs2017.vs_ent` directory, `\metabox_downloads\vs2017.vs_ent\zip` size is just 12KB.
Log: https://gist.github.com/shurick81/704b89af773109fa1fb01e0c43403761
I looked thoroughly at the process and noticed a separate temporary window, apart from console, and error there: `The package 'Microsoft.VisualStudio.Workload.ManagedDesktopBuildTools' is not recognized.`

I will continue troubleshooting and installing VS in corresponding branch.

**Comment by shurick81** [comment]
This VS issue looks relevant: https://github.com/MicrosoftDocs/visualstudio-docs/issues/1088

**Comment by avishnyakov** [comment]
Implemented in the latest integration, waiting confirmation from regression runs. @shurick81 , you might want to give it a try too

`
metabox vagrant up contoso16::vs17
`
